lib/index.ts

```diff
@@ -486,20 +503,27 @@ export async function buildDepTreeFromImportsAndModules(root: string = '.', targ
   const localPackageWithMainModule = localPackages
       .find((localPackage) => !!(localPackage.Module && localPackage.Module.Main));
   if (localPackageWithMainModule && localPackageWithMainModule!.Module!.Path) {
-    depTree.name = localPackageWithMainModule!.Module!.Path;
+    depGraphBuilder = new DepGraphBuilder({ name: 'gomodules' }, {
+      name: localPackageWithMainModule!.Module!.Path,
+      version: projectVersion,
+    });
   }
-
   const topLevelDeps = extractAllImports(localPackages);
-  buildTree(depTree, topLevelDeps, packagesByName);
-  return depTree;
+  
+  const childrenChain = new Map();
+  const ancestorsChain = new Map();
+
+  buildGraph(depGraphBuilder, topLevelDeps, packagesByName, 'root-node', childrenChain, ancestorsChain);
+
+  return depGraphBuilder.build();
 }
 
-function buildTree(
-  depTreeNode: DepTree,
+function buildGraph(depGraphBuilder: DepGraphBuilder,
   depPackages: string[],
-  packagesByName: GoPackagesByName,
-) {
+  packagesByName: GoPackagesByName, currentParent: string, childrenChain: Map<string, Array<string>>, ancestorsChain: Map<string, Array<string>>) {
+  
   const depPackagesLen = depPackages.length;
+
   for (let i = depPackagesLen - 1; i >= 0; i--) {
```

Comment:
is reversing over array important here ?

Reply:
reverse loops are faster